File: src/weed_api/species.py
"""Metadatos de las clases DeepWeeds (binomiales según Olsen et al. 2019).

El orden replica el orden canónico de `weed_api.labels.LABELS`.
Los nombres comunes en español solo se incluyen cuando son verificables
en Argentina; no se inventan nombres vernáculos.
"""

from weed_api.api.schemas import SpeciesInfo

SPECIES: list[SpeciesInfo] = [
    SpeciesInfo(
        label="Chinee apple",
        scientific="Ziziphus mauritiana",
        blurb="Árbol espinoso de fruto pequeño; invasor agresivo de pasturas tropicales.",
    ),
    SpeciesInfo(
        label="Lantana",
        scientific="Lantana camara",
        common_es="Lantana",
        blurb="Arbusto de flores multicolores; tóxica para el ganado e invasora de pasturas.",
    ),
    SpeciesInfo(
        label="Parkinsonia",
        scientific="Parkinsonia aculeata",
        common_es="Cina-cina",
        blurb="Árbol espinoso de ramas verdes; forma matorrales densos junto al agua.",
    ),
    SpeciesInfo(
        label="Parthenium",
        scientific="Parthenium hysterophorus",
        blurb="Hierba anual de flores blancas pequeñas; alergénica y muy competitiva.",
    ),
    SpeciesInfo(
        label="Prickly acacia",
        scientific="Vachellia nilotica",
        blurb="Acacia espinosa; invade pastizales y compite con las forrajeras.",
    ),
    SpeciesInfo(
        label="Rubber vine",
        scientific="Cryptostegia grandiflora",
        blurb="Enredadera de látex tóxico; cubre y asfixia la vegetación nativa.",
    ),
    SpeciesInfo(
        label="Siam weed",
        scientific="Chromolaena odorata",
        blurb="Arbusto de crecimiento muy rápido; forma matas densas que desplazan pasturas.",
    ),
    SpeciesInfo(
        label="Snake weed",
        scientific="Stachytarpheta spp.",
        blurb="Hierba de espigas azul-violáceas; común en pasturas degradadas.",
    ),
    SpeciesInfo(
        label="Negative",
        blurb="Ninguna de las 8 malezas del modelo.",
    ),
]

File: frontend/src/data/species.ts
/**
 * Metadatos de las 8 especies DeepWeeds (binomiales según Olsen et al. 2019).
 * `id` es la etiqueta exacta del modelo (src/weed_api/labels.py).
 * `es` es el titular: nombre común verificado en Argentina o, si no existe,
 * el nombre científico — no se inventan nombres vernáculos.
 */

export interface Species {
  id: string;
  es: string;
  scientific: string;
  blurb: string;
  image: string;
}

export const PLACEHOLDER_IMAGE = "/species/placeholder.svg";

export const SPECIES: Species[] = [
  {
    id: "Chinee apple",
    es: "Ziziphus mauritiana",
    scientific: "Ziziphus mauritiana",
    blurb: "Árbol espinoso de fruto pequeño; invasor agresivo de pasturas tropicales.",
    image: "/species/chinee-apple.jpg",
  },
  {
    id: "Lantana",
    es: "Lantana",
    scientific: "Lantana camara",
    blurb: "Arbusto de flores multicolores; tóxica para el ganado e invasora de pasturas.",
    image: "/species/lantana.jpg",
  },
  {
    id: "Parkinsonia",
    es: "Cina-cina",
    scientific: "Parkinsonia aculeata",
    blurb: "Árbol espinoso de ramas verdes; forma matorrales densos junto al agua.",
    image: "/species/parkinsonia.jpg",
  },
  {
    id: "Parthenium",
    es: "Parthenium hysterophorus",
    scientific: "Parthenium hysterophorus",
    blurb: "Hierba anual de flores blancas pequeñas; alergénica y muy competitiva.",
    image: "/species/parthenium.jpg",
  },
  {
    id: "Prickly acacia",
    es: "Vachellia nilotica",
    scientific: "Vachellia nilotica",
    blurb: "Acacia espinosa; invade pastizales y compite con las forrajeras.",
    image: "/species/prickly-acacia.jpg",
  },
  {
    id: "Rubber vine",
    es: "Cryptostegia grandiflora",
    scientific: "Cryptostegia grandiflora",
    blurb: "Enredadera de látex tóxico; cubre y asfixia la vegetación nativa.",
    image: "/species/rubber-vine.jpg",
  },
  {
    id: "Siam weed",
    es: "Chromolaena odorata",
    scientific: "Chromolaena odorata",
    blurb: "Arbusto de crecimiento muy rápido; forma matas densas que desplazan pasturas.",
    image: "/species/siam-weed.jpg",
  },
  {
    id: "Snake weed",
    es: "Stachytarpheta spp.",
    scientific: "Stachytarpheta spp.",
    blurb: "Hierba de espigas azul-violáceas; común en pasturas degradadas.",
    image: "/species/snake-weed.jpg",
  },
];

export function speciesByLabel(label: string): Species | undefined {
  return SPECIES.find((s) => s.id === label);
}
